lib/core/src/Cardano/Wallet.hs

```diff
@@ -1288,11 +1288,15 @@ selectAssetsNoOutputs ctx wid tx transform = do
                     (view #tokens (head $ outputsCovered sel))
                     (TokenBundle.fromCoin deposit)
             in
-                once (TokenBundle.add reclaim) (changeGenerated sel)
+                once
+                    (TokenBundle.empty)
+                    (TokenBundle.add reclaim)
+                    (changeGenerated sel)
         }
   where
-    once :: (a -> a) -> NonEmpty a -> NonEmpty a
-    once fn (a :| as) = fn a :| as
+    once :: a -> (a -> a) -> [a] -> [a]
+    once _ f (a : as) = f a : as
+    once a _ [] = [a]
```

Comment:
This needs checking.

Reply:
It seems a bit unnatural to have `changeGenerated = [mempty]`. Why not `[]`?

Reply:
This is wrong indeed. The name of the function was maybe confusing. This is what this piece of code is intended to do:

```hs
                reclaim `addToHead` changeGenerated sel
  where
    addToHead :: TokenBundle -> [TokenBundle] -> [TokenBundle]
    addToHead x []    = [x]
    addToHead x (h:q) = TokenBundle.add x h : q
```